Remove the `assert` lib

nandito opened the pull request:
* Bump jslib-media, the new version:
    * Adds media device and constraints helpers
    * Replaces the 3rd party `assert` lib
* Import `assert` from jslib-media and remove the [assert](https://www.npmjs.com/package/assert) lib from browser-sdk's dependencies.
* Remove some of the redundant type checks.

The `assert` lib replacement should fix the "Could not find module in path: 'assert/assert.js'" error in CodeSandbox.

### Test plan

1. Install the new browser-sdk with different bundlers/frameworks (cra, vite, etc)
2. Make sure you don't get the "Could not find module" error for `assert`.

### Related PR

https://github.com/whereby/jslib-media/pull/32

nandito committed:
Bump jslib-media version (minor)

This new version:
* Adds media device and constraints helpers
* Replaces the 3rd party `assert` lib

The `assert` lib replacement should fix the "Could not find module in
path: 'assert/assert.js'" error in CodeSandbox.

thyal reviewed (commented):
Nice. But we also use `assert` directly in this repo - maybe we can replace the usage in this PR as well?

nandito commented:
> Nice. But we also use `assert` directly in this repo - maybe we can replace the usage in this PR as well?

Yeah I just realized this now. Actually we use it way more here than in jslib-media. 🙈
I'm going to remove it from here too. 🚧 👷‍♂️

kevinwhereby commented:
> Actually we use it way more here than in jslib-media. 🙈

I reckon a lot of it is reduntant with the move to typescript in the repo, so my vote would be we remove calls to it where possible as well.

I.e. all the `assertOk(typeof thing === "thingtype")`

nandito committed:
Import `assert` from jslib-media and remove 3rd party assert

nandito committed:
Remove redundant type checks

TypeScript already checks the types, so no need to explicitly re-check
them with `assert.ok` or `assertString`.

nandito commented:
> > Actually we use it way more here than in jslib-media. 🙈
> 
> I reckon a lot of it is redundant with the move to typescript in this repo, so my vote would be we remove calls to it where possible as well.
> 
> I.e. all the `assertOk(typeof thing === "thingtype")`

I agree! I have removed some now, but there are still quite a few remaining. I'm not sure if I want to increase the scope, as the original task was to remove the 3rd party `assert` and I have already done that. Therefore, bundlers, tools, and frameworks should no longer complain about the missing module.

thyal reviewed (approved):
🍰 Works nicely.
